controller/controller_server/utils.py

```diff
@@ -154,14 +153,48 @@ def validate_delete_snapshot_request(request):
     logger.debug("request validation finished.")
 
 
+def validate_validate_volume_capabilities_request(request):
+    logger.debug("validating validate_volume_capabilities volume request")
+
+    logger.debug("validating volume id")
+    volume_id = request.volume_id
+    if not request.volume_id:
+        raise ValidationException(messages.name_should_not_be_empty_message)
+    if config.PARAMETERS_OBJECT_ID_DELIMITER not in volume_id:
+        raise ObjectIdError(config.OBJECT_TYPE_ID_VOLUME, volume_id)
+
+    logger.debug("validating volume capabilities")
+    if not request.volume_capabilities:
+        raise ValidationException(messages.capabilities_not_set_message)
+    validate_csi_volume_capabilties(request.volume_capabilities)
+
+    logger.debug("validating secrets")
+    if request.secrets:
```

Comment:
is this `if` really necessary?

Reply:
why not?

Reply:
what happens if we remove them here?

Reply:
secret is not mandatory filed so if we wont get it in request and still test it we will get error of checking and comparing None..

Reply:
I'm not sure I understand. in which line would we get an error?

Reply:
inside testing secret

Reply:
we need to test it due to the fact its a possible field in the request

Reply:
I could find the line `inside testing secret` - could you please provide a number/link with explanation?

Reply:
ok I will be more specific, it will fail [here](https://github.com/IBM/ibm-block-csi-driver/blob/8cc238debededf08f16d0279b1abcb952672c6a9/controller/controller_server/utils.py#L45)

Reply:
then why not just remove the `else` you linked?

Reply:
I removed the if.

Reply:
thanks. could you also remove it from the rest of the places that call `validate_secret`?

Reply:
done